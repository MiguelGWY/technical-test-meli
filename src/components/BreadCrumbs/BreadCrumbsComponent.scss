.container-breadcrumbs {
  &__breadcrumb {
    font-size: 14px;
    padding: 0;
  }

  &__list {
    list-style: none;
    display: inline-block;
    color: gray;
  }

  &__redirect-button {
    color: gray;
    text-decoration: none;
    border: none;
    background: none;
    cursor: pointer;
  }
}
