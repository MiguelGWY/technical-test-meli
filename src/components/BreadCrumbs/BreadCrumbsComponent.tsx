import { useNavigate } from "react-router-dom";
import "./BreadCrumbsComponent.scss";

export const BreadCrumbs = ({ categories }: BreadCrumbsProps) => {
  const navigate = useNavigate();
  const nameSpace = "container-breadcrumbs";

  if (!categories || categories.length === 0) return null;

  return (
    <div data-testid="container-breadcrumbs" className={nameSpace}>
      <ol className={`${nameSpace}__breadcrumb`}>
        {categories.map((crumb, index) => (
          <li className={`${nameSpace}__list`} key={crumb}>
            <button
              className={`${nameSpace}__redirect-button`}
              onClick={() => navigate(`/items?search=${crumb}`)}
            >
              {crumb}
            </button>
            {index < categories.length - 1 && " > "}
          </li>
        ))}
      </ol>
    </div>
  );
};
