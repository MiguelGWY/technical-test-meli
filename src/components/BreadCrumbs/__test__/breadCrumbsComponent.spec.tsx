import { render } from "@testing-library/react";
import { BreadCrumbs } from "../BreadCrumbsComponent";
import "@testing-library/jest-dom";

const categories = ["Category 1", "Category 2"];

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("BreadCrumbsComponent", () => {
  it("should render BreadCrumbsComponent", () => {
    const { container } = render(<BreadCrumbs categories={categories} />);
    expect(container).toMatchSnapshot();
  });

  it('should render a div with class "container-breadcrumbs" when categories prop is not empty', () => {
    const categories = ["category1", "category2"];

    const { getByTestId } = render(<BreadCrumbs categories={categories} />);

    const container = getByTestId("container-breadcrumbs");

    expect(container).toBeInTheDocument();
    expect(container).toHaveClass("container-breadcrumbs");
  });

  it("should render null when categories prop is empty", () => {
    const categories: string[] = [];

    const { container } = render(<BreadCrumbs categories={categories} />);
    expect(container).toBeEmptyDOMElement();
  });
});
