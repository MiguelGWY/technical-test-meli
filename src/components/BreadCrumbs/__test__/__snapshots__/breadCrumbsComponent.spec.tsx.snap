// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbsComponent should render BreadCrumbsComponent 1`] = `
<div>
  <div
    class="container-breadcrumbs"
    data-testid="container-breadcrumbs"
  >
    <ol
      class="container-breadcrumbs__breadcrumb"
    >
      <li
        class="container-breadcrumbs__list"
      >
        <button
          class="container-breadcrumbs__redirect-button"
        >
          Category 1
        </button>
         &gt; 
      </li>
      <li
        class="container-breadcrumbs__list"
      >
        <button
          class="container-breadcrumbs__redirect-button"
        >
          Category 2
        </button>
      </li>
    </ol>
  </div>
</div>
`;
